### PR TITLE
build(dcs/maintainwindow): upgrade the API of DCS maintainwindows to V2 and move code files to the service path

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

- Upgrade the API of DCS maintainwindows to V2.
- Move code files to the service path.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run TestAccDcsMaintainWindowDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run TestAccDcsMaintainWindowDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsMaintainWindowDataSource_basic
=== PAUSE TestAccDcsMaintainWindowDataSource_basic
=== CONT  TestAccDcsMaintainWindowDataSource_basic
--- PASS: TestAccDcsMaintainWindowDataSource_basic (53.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       54.056s

```

### CODE REVIEW

File: huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_maintainwindow_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+				),
+			},
+		},
+	})
+}
+
+var testAccDcsMaintainWindowDataSource_basic = fmt.Sprintf(`
+data "huaweicloud_dcs_maintainwindow" "maintainwindow1" {
+seq = 1
+default = false
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_maintainwindow_test.go]: format error

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go [hunk 2]
```diff
@@ -45,20 +47,20 @@ func DataSourceDcsMaintainWindowV1() *schema.Resource {
 	}
 }
 
-func dataSourceDcsMaintainWindowV1Read(d *schema.ResourceData, meta interface{}) error {
+func dataSourceDcsMaintainWindowRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go]: suggest to use _ if ctx is not used

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go [hunk 3]
```diff
@@ -45,20 +47,20 @@ func DataSourceDcsMaintainWindowV1() *schema.Resource {
...
 	config := meta.(*config.Config)
-	dcsV1Client, err := config.DcsV1Client(GetRegion(d, config))
+	dcsV2Client, err := config.DcsV2Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating dcs key client: %s", err)
+		return fmtp.DiagErrorf("Error creating dcs key client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go]: s/dcs key/DCS

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go [hunk 4]
```diff
@@ -45,20 +47,20 @@ func DataSourceDcsMaintainWindowV1() *schema.Resource {
...
-	v, err := maintainwindows.Get(dcsV1Client).Extract()
+	v, err := maintainwindows.Get(dcsV2Client).Extract()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	maintainWindows := v.MaintainWindows
-	var filteredMVs []maintainwindows.MaintainWindow
+	filteredMVs := make([]maintainwindows.MaintainWindow, 0, len(maintainWindows))
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go]: are you interesting to try `utils.FliterSliceWithField`?

File: huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_maintainwindow_test.go (hunk 1 above)
Comment (chengxiangdong) [reply huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_maintainwindow_test.go]: done

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go (hunk 3 above)
Comment (chengxiangdong) [reply huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go]: done

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go (hunk 4 above)
Comment (chengxiangdong) [reply huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go]: Replaced, now it more readable

**Pull request merged**